### PR TITLE
Add Time-optimal Trajectory Generation tutorial

### PR DESCRIPTION
**Pull request by henningkayser**
This PR adds a short description for Time-optimal Trajectory Generation ([#1365](https://github.com/ros-planning/moveit/pull/1365), [#809](https://github.com/ros-planning/moveit/pull/809)).

### CODE REVIEW

File: doc/time_parameterization/time_parameterization_tutorial.rst [hunk 1]
```diff
@@ -16,8 +16,10 @@ The speed of a parameterized kinematic trajectory can also be modified during ru
...
+MoveIt! can support different algorithms for post-processing a kinematic trajectory to add timestamps and velocity/acceleration values. Currently there are three available by default in MoveIt!: `Iterative Parabolic Time Parameterization <https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_core/trajectory_processing/src/iterative_time_parameterization.cpp>`_, `Iterative Spline Parameterization <https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp>`_, and `Time-optimal Trajectory Generation <https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp>`__.
```
Comment (davetcoleman) [inline doc/time_parameterization/time_parameterization_tutorial.rst]: i think this should be a bulleted list now

File: doc/time_parameterization/time_parameterization_tutorial.rst [hunk 2]
```diff
@@ -16,8 +16,10 @@ The speed of a parameterized kinematic trajectory can also be modified during ru
...
 
-The Iterative Spline Parameterization algorithm was recently merged into mainstream MoveIt! in `PR 382 <https://github.com/ros-planning/moveit/pull/382>`_ to deal with these issues. While preliminary experiments are very promising, we are waiting for more feedback from the community before replacing the Iterative Parabolic Time Parameterization algorithm completely.
\ No newline at end of file
+The Iterative Spline Parameterization algorithm was recently merged into mainstream MoveIt! in `PR 382 <https://github.com/ros-planning/moveit/pull/382>`_ to deal with these issues. While preliminary experiments are very promising, we are waiting for more feedback from the community before replacing the Iterative Parabolic Time Parameterization algorithm completely.
```
Comment (davetcoleman) [inline doc/time_parameterization/time_parameterization_tutorial.rst]: remove the outdated wording 'recently merged'

File: doc/time_parameterization/time_parameterization_tutorial.rst [hunk 3]
```diff
@@ -16,8 +16,10 @@ The speed of a parameterized kinematic trajectory can also be modified during ru
...
+
+Time-optimal Trajectory Generation introduced in PRs `#809 <https://github.com/ros-planning/moveit/pull/809>`_ and `#1365 <https://github.com/ros-planning/moveit/pull/1365>`_ produces trajectories with very smooth and continuous velocity profiles. The method is based on fitting path segments to the original trajectory and then sampling new waypoints from the optimized path. This is different from strict time parameterization methods as resulting waypoints may divert from the original trajectory within a certain tolerance. As a consequence, additional collision checks might be required when using this method.
```
Comment (davetcoleman) [inline doc/time_parameterization/time_parameterization_tutorial.rst]: off topic, but we should make one of these the default in master. but i think we need more testing sigh

**Comment by rhaschke** [comment]
Ping @henningkayser.

**Comment by rhaschke** [comment]
@henningkayser, are you sure about the syntax `:moveit_codedir:name<file.cpp>`? This [displays weird in github](https://github.com/ros-planning/moveit_tutorials/blob/a067f2f1f606005cd7a6bf9c03ac937f16835784/doc/time_parameterization/time_parameterization_tutorial.rst#time-parameterization-algorithms).

**Comment by henningkayser** [comment]
> @henningkayser, are you sure about the syntax `:moveit_codedir:name<file.cpp>`? This [displays weird in github](https://github.com/ros-planning/moveit_tutorials/blob/a067f2f1f606005cd7a6bf9c03ac937f16835784/doc/time_parameterization/time_parameterization_tutorial.rst#time-parameterization-algorithms).

It follows the pattern introduced here https://github.com/ros-planning/moveit_tutorials/commit/c4efe287fdf326992c536dca55fc6daf5975b158#diff-7e952a5b50d405ec0b08afe9cb45e971 which also has caused conflicts with the PR branch. Actually at first I was just as confused as you, but it's apparently a common feature of Sphinx.

**Pull request merged**